apps/judicial-system/web/src/components/Table/Table.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { ReactNode, useContext, useMemo } from 'react'
+import { useIntl } from 'react-intl'
+import { useLocalStorage } from 'react-use'
+import parseISO from 'date-fns/parseISO'
+import { AnimatePresence } from 'framer-motion'
+
+import { Box, Text } from '@island.is/island-ui/core'
+import { theme } from '@island.is/island-ui/theme'
+import { formatDate } from '@island.is/judicial-system/formatters'
+import { isDistrictCourtUser } from '@island.is/judicial-system/types'
+import { core } from '@island.is/judicial-system-web/messages'
+
+import { CaseListEntry, CaseState } from '../../graphql/schema'
+import MobileCase from '../../routes/Shared/Cases/MobileCase'
+import { directionType, sortableTableColumn, SortConfig } from '../../types'
+import { useCase, useCaseList, useViewport } from '../../utils/hooks'
+import { compareLocaleIS } from '../../utils/sortHelper'
+import ContextMenu, {
+  ContextMenuItem,
+  type MenuItems,
+  PrebuiltMenuItems,
+  useContextMenu,
+} from '../ContextMenu/ContextMenu'
+import IconButton from '../IconButton/IconButton'
+import { UserContext } from '../UserProvider/UserProvider'
+import SortButton from './SortButton/SortButton'
+import { table as strings } from './Table.strings'
+import * as styles from './Table.css'
+
+interface TableProps {
+  thead: {
+    title: string
+    sortable?: {
+      isSortable: boolean
+      key: sortableTableColumn
+    }
+  }[]
+  data: CaseListEntry[]
+  columns: { cell: (row: CaseListEntry) => ReactNode }[]
+  contextMenu?: {
+    menuItems: MenuItems
+  }
+  onDeleteCase?: (theCase: CaseListEntry) => void
```

Comment:
Discussed offline as well, but I think we should try to keep the Table component as flexible as possible, so I think we need to avoid specific things like this. I had an idea of an alternative way where we can pass a function that generates the items in as a prop rather than the actual menu items. Alternatively, we could pass the entire ContextMenu in as a render function, I think just passing in a generation of menu items is enough though for now. Here's my thinking: https://github.com/island-is/island.is/commit/339c5491712e45b740a2711581d858fd6c710f1e